sound/soc/sof/virtio-miscdev.c

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2017 Intel Corporation. All rights reserved.
+ *
+ *  Author: Libin Yang <libin.yang@intel.com>
+ *	  Luo Xionghu <xionghu.luo@intel.com>
+ *	  Liam Girdwood <liam.r.girdwood@linux.intel.com>
+ */
+
+#include <linux/module.h>
+#include <linux/miscdevice.h>
+#include <linux/fs.h>
+#include <linux/file.h>
+#include <linux/slab.h>
+
+#include "sof-priv.h"
+#include "virtio-miscdev.h"
+#include <linux/vbs/vbs.h>
+
+/*
+ * This module registers a device node /dev/vbs_k_audio,
+ * that handle the communication between Device Model and
+ * the virtio backend service. The device model can
+ * control the backend to : set the status, set the vq account and etc.
+ * The config of the DM and VBS must be accordance.
+ */
+
+static struct virtio_miscdev *virtio_audio;
+
+static struct virtio_miscdev *get_virtio_audio(void)
+{
+	return virtio_audio;
+}
+
+void *get_sof_dev(void)
+{
+	struct virtio_miscdev *vaudio = get_virtio_audio();
+
+	if (vaudio)
+		return vaudio->priv;
+
+	return NULL;
+}
+
+/* To be implemented later for sof related */
+static int sof_virtio_open(struct file *f, void *data)
+{
+	return 0;
+}
+
+/* To be implemented later for sof related */
+static long sof_virtio_ioctl(struct file *f, void *data, unsigned int ioctl,
+			     unsigned long arg)
+{
+	return 0;
+}
+
+/* To be implemented later for sof related */
+static int sof_virtio_release(struct file *f, void *data)
+{
+	return 0;
+}
+
+int snd_sof_virtio_miscdev_register(struct snd_sof_dev *sdev)
+{
+	struct virtio_miscdev *vaudio;
+	int ret;
+
+	ret = snd_audio_virtio_miscdev_register(sdev->dev, sdev, &vaudio);
+	if (ret)
+		return ret;
+
+	vaudio->open = sof_virtio_open;
+	vaudio->ioctl = sof_virtio_ioctl;
+	vaudio->release = sof_virtio_release;
+
+	return 0;
+}
+
+int snd_sof_virtio_miscdev_unregister(void)
```

Comment:
needed ?

Reply:
Do you mean snd_sof_virtio_miscdev_unregister? For module mode, I think yes, we need it when unload the sof modules.